### PR TITLE
TMA-45: add test for MdIdToUri mixin

### PR DESCRIPTION
**Pull request by hacxman**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/329/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/57/ : FAILURE in 18s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/367/ : FAILURE in 16s

**Comment by panjan** [comment]
ok to test

File: spec/unit/mixins/id_to_uri_spec.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# LICENSE file in the root directory of this source tree.
+
+require 'gooddata/mixins/md_id_to_uri'
+
+describe GoodData::Mixin::MdIdToUri do
+  before :each do
+    @client = ConnectionHelper.create_default_connection
+    project = ProjectHelper.get_default_project(client: @client)
+
+    class SomeTestClass
```
Comment (yenkins) [inline spec/unit/mixins/id_to_uri_spec.rb]: Has no descriptive comment (IrresponsibleModule)

File: spec/unit/mixins/id_to_uri_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      extend GoodData::Mixin::MdIdToUri
+    end
+
+    @opts = {client: @client, project: project}
```
Comment (yenkins) [inline spec/unit/mixins/id_to_uri_spec.rb]: Style/SpaceInsideHashLiteralBraces: Space inside { missing.

Comment (yenkins) [inline spec/unit/mixins/id_to_uri_spec.rb]: Style/SpaceInsideHashLiteralBraces: Space inside } missing.

File: spec/unit/mixins/id_to_uri_spec.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  end
+
+  it 'should throw BadRequest for -1' do
+    expect do
+      SomeTestClass.identifier_to_uri(@opts, -1)
+    end.to raise_error(RestClient::BadRequest)
+  end
+
+  it 'should return nil for unknown id' do
+    expect(SomeTestClass::identifier_to_uri(@opts, 0)).to be_nil
```
Comment (yenkins) [inline spec/unit/mixins/id_to_uri_spec.rb]: Style/ColonMethodCall: Do not use `::` for method calls.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/330/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/60/ : FAILURE in 35s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/370/ : FAILURE in 35s

**Pull request closed without merging**